react/ProductDescription.tsx

```diff
@@ -82,7 +82,7 @@ function ProductDescription(props: PropsWithChildren<Props>) {
 
       <div className={`${handles.productDescriptionText} c-muted-1`}>
         {collapseContent ? (
-          <GradientCollapse collapseHeight={220}>
+          <GradientCollapse collapseHeight={220} product={product}>
```

Comment:
I don't think it's a good idea to attach the `GradientCollapse` component to the product. The component has a `onCollapsedChange` prop. You can send the push through it.

This way, you keep the product logic here